src/libraries/Common/src/System/Number.NumberBuffer.cs

```diff
@@ -31,7 +31,7 @@ internal static partial class Number
             public NumberBufferKind Kind;
             public byte* DigitsPtr;
             public int DigitsLength;
-            public readonly Span<byte> Digits => new Span<byte>(DigitsPtr, DigitsLength);
+            public readonly Span<byte> Digits => MemoryMarshal.CreateSpan(ref *DigitsPtr, DigitsLength);
```

Comment:
This is because the Span ctor has a length < 0 check and MemoryMarshal.CreateSpan doesn't?

Reply:
Yes - it seems to be the cause based on the regression & the code I'd changed.